randovania/games/prime_hunters/exporter/patch_data_factory.py

```diff
@@ -52,16 +61,53 @@ def force_field_index_for_requirement(game: GameDescription, requirement: Layout
     "Octolith8": 7,
 }
 
+_ITEM_TO_ITEM_TYPE = {
+    "Energy Tank": 4,
+    "Volt Driver": 5,
+    "Missile": 6,
+    "Battlehammer": 7,
+    "Imperialist": 8,
+    "Judicator": 9,
+    "Magmaul": 10,
+    "Shock Coil": 11,
+}
+
 
 class HuntersPatchDataFactory(PatchDataFactory[HuntersConfiguration, HuntersCosmeticPatches]):
     def game_enum(self) -> RandovaniaGame:
         return RandovaniaGame.METROID_PRIME_HUNTERS
 
-    def _calculate_starting_inventory(self, resources: ResourceCollection) -> dict[str, str]:
-        result = {
-            # TODO: Take starting weapons into account
-            "weapons_string": "00000101",
-        }
+    def _calculate_starting_inventory(self, resources: ResourceCollection) -> dict[str, str | int]:
+        result: dict[str, str | int] = {}
+        starting_items = {}
+
+        for resource, quantity in resources.as_resource_gain():
+            try:
+                starting_items[item_type_for_item_resource(resource)] = quantity
+            except KeyError:
+                print(f"Skipping {resource} for starting inventory: no item id")
+                continue
```

Comment:
I'm not sure what the advantage of doing this is when you only ever index it by the name of the item. couldn't it just be like:
```py
starting_items = defaultdict(int)
starting_items.update({resource.long_name: quantity for resource, quantity in resources.as_resource_gain()})
```